### ISSUE TITLE
looking at mirrors and portals makes water in the world visible through solid blocks

### ISSUE THREAD

**Issue by PowerMilian** [qouteall/ImmersivePortalsMod#886]
i have the same issue as this one #605 but for me its when i look at a mirror or portal.
im playing on the all of fabric 3 modpack and with the immersive-portals-0.101-mc1.16.5 mod version
(same with / without fabolous settings)
![image](https://user-images.githubusercontent.com/76727746/143498276-58493930-e513-4d3c-9db7-75e8362b88c3.png)
![image](https://user-images.githubusercontent.com/76727746/143498394-5dfd8181-2ca3-4b43-8e75-e309c4378026.png)

**Comment by qouteall** [comment]
Does this issue occur with only Immersive Portals and Fabric API? (It's recommended to test it in a new world)

**Comment by qouteall** [comment]
according to https://github.com/qouteall/ImmersivePortalsMod/issues/801 , may be incompatible with rolls and modern industrial

**Comment by CSdiscoveries** [comment]
I believe I am having this exact issue, and only when looking at a portal. If the portal's surface is not visible, there is almost no issue; however, as soon as the surface comes into view, a few things begin to happen:
1. The framerate of the game freezes noticeably, but quickly returns to normal (though it periodically keeps freezing while the portal is visible)
2. I start to see certain blocks through terrain, like water and Corundum blocks (from Quark). As soon as I move enough so that no part of the portal is being rendered, this stops happening; if I look back it happens again, and so on. I believe it also happens with lava, too.
3. After a while, there appears to be an odd issue with Distant Horizons, where the fake chunks it renders are shifted upwards by a noticeable amount. This doesn't stop even when I look away from the portal.
The Framerate and Distant Horizons issues are not important here, and would be better suited to being their own issues. However, this is the only place where I see the improper rendering of water (in this specific way) being reported. I do not know what water and corundum have in common that would cause them both to be affected in the same way, but that information may help in order to determine the root cause. As well, as opposed to the original poster, I am using the mod on Minecraft 1.18.2, with Forge as opposed to Fabric.
In addition, after looking at issue 801, this appears to be something entirely separate from that. And, I don't seem to be able to find any mods named "rolls" or "modern industrial". I certainly don't have them installed, unless those are the mistranslated names of mods that I do have (which could be possible.) However, on the topic of incompatibilities, I do have Flywheel, Rubidium, and BetterFPS installed, at least one of which is definitely considered "incompatible". However, there is one other possibility that I can think of for an incompatibility: "Amplified Nether". Since it changes the Nether's height significantly, it may be a factor in the many bugs I experienced.

tl;dr: I have this problem too, in a newer MC version and on Forge. I am able to see Quark's Corundum blocks, Water, and Lava through the ground or other solid blocks.

If you need any more info from me, let me know. I've tried to be thorough here, but nobody is infallible. The one thing that I can think of adding to this, but really don't want to, would be my rather long mod list. I doubt this comment needs to be any longer.

**Comment by qouteall** [comment]
@CSdiscoveries 
Firstly this mod is incompatible with Distant Horizons. 

Second, don't use Fabulous graphics. Use Fancy. Because Fabulous mod uses another set of slower and complex thing to render transparent things which may be incompatibel with immptl.

**Comment by CSdiscoveries** [comment]
At the time of me installing this mod (and of finding the bug), my graphics were set to fast. My hardware does not support Fabulous graphics. For testing purposes, I just now changed the graphics settings to Fancy; there didn't seem to make any difference, aside from that I started to be able to see the bug in question even while the portal was not being rendered.

**Comment by qouteall** [comment]
@CSdiscoveries
Fast and Fancy are both fine. 
Does this issue occur with only immersive portals mod and fabric API? If not, please find which mod is incompatible by removing half of the mods and test repeatedly.

**Comment by CSdiscoveries** [comment]
In both Fabric and Forge, I am not noticing the issue if Immersive Portals and Fabric API are the only mods installed. It will take me a bit to see what specific mods cause issues, since I tend to use a lot of mods. I will comment here again once I get some results.

**Comment by CSdiscoveries** [comment]
The issue was pretty easy to find: Like stated previously, Distant Horizons is incompatible. It seems to be the only reason for that glitch. Removing it causes everything to work properly, minus some momentary stuttering/lag if the portal is being rendered on-screen. I don't know exactly why the rendering changes like that, since I think DH is only supposed to affect things that are far away. I tested in both Fabric and Forge, and the issue was the same in both environments. I don't know why this specific incompatibility happens, or even whose end the issue is on, but hopefully that information has been somewhat helpful. 

Thank you for your time here; while I'd like to use this mod's functionality in survival, it's unfortunately too performance-heavy for me to choose it over DH. But, I do appreciate the effort towards making it work.

**Comment by qouteall** [comment]
Close because outdated.
